torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -359,6 +360,14 @@ class CudaKernelGenerator : private OptOutConstDispatch {
     bool is_vector_op = false;
     size_t vector_word_size = 1;
 
+    if (uop->out()->isA<kir::TensorIndex>()) {
```

Comment:
I believe this is an op to initialize the output tensor of an MmaOp.

The condition isn't very intuitive to me. It uses the fact that the definition of the tensor has the original MmaOp expression rather than its initialization, which is true but seems like an implementation detail. 

Wouldn't it be more robust to look at the parallel type of IterDomains? There should be an IterDomain with `ParallelType::Mma`, and isn't it be possible to use that info to determine if this is an initialization op of a Mma output tensor? I believe that's how we generate vectorized code for UnaryOps with vectorized IterDomains.

Reply:
Yes this is detection of the initialization for mma ops. Which is now implemented via specialized runtime functions. 

I have changed the detection of this to using the parallel type.